src/app/admin/page.tsx

```diff
@@ -1,60 +1,251 @@
+                Change the role of a user
+              </p>
+              <div className="my-4 rounded-2xl border-2 p-4">
+                <div className="rounded-md bg-white p-4">
+                  <input
+                    type="text"
+                    className="mb-3 w-full rounded-md border p-2"
+                    placeholder="Search for a user..."
+                    value={state.searchTermUsers}
+                    onChange={(e) =>
+                      setState((prevState) => ({
+                        ...prevState,
+                        searchTermUsers: e.target.value,
+                      }))
+                    }
+                  />
+                  <ul className="class-list max-h-60 overflow-y-auto">
+                    {users!.map((u) => (
+                      <li
+                        key={u.uid}
+                        className="grid cursor-pointer grid-cols-1 gap-4 rounded-md border p-4 text-center shadow-md hover:bg-gray-200 md:grid-cols-2 lg:grid-cols-3"
+                        onClick={() => openModal(u)} // Open modal on click
+                      >
+                        <div className="hidden font-bold lg:block">
+                          {u.displayName}
+                        </div>
+                        <div className="font-normal md:font-bold lg:font-normal">
+                          {u.email}
+                        </div>
+                        <div className="font-bold">{u.access}</div>
+                      </li>
+                    ))}
+                  </ul>
+                </div>
+              </div>
+            </>
+          )}
+
+          <p className="w-full text-pretty text-lg opacity-70 sm:text-lg lg:text-xl">
+            Select AP Class for Title
+          </p>
+          <div className="my-4 rounded-2xl border-2 p-4">
+            <div className="rounded-md bg-white p-4">
+              <input
+                type="text"
+                className="mb-3 w-full rounded-md border p-2"
+                placeholder="Search for a class..."
+                value={state.searchTermAPClasses}
+                onChange={(e) =>
+                  setState((prevState) => ({
+                    ...prevState,
+                    searchTermAPClasses: e.target.value,
+                  }))
+                }
+              />
+              <ul className="class-list max-h-60 overflow-y-auto">
+                {filteredClasses.map((apClass: string) => (
+                  <li
+                    key={apClass}
+                    className="cursor-pointer rounded-md p-2 hover:bg-gray-200"
+                    onClick={() =>
+                      router.push(
+                        `/admin/subject/${apClass
+                          .replace(/AP /g, "")
+                          .toLowerCase()
+                          .replace(/[^a-z1-9 ]+/g, "")
+                          .replace(/\s/g, "-")}`,
```

Comment:
Ideally this regex should be in a function named `formatAPClassName` (or something) so it can be reused easily

Reply:
brug